docs/implementation.md

```diff
@@ -0,0 +1,210 @@
+## Implementing Links
+
+Before implementing your own Link, be sure to read the [usage](usage.md) document.
+
+Links implement a `request` method that takes an operation and returns the results in an Observable.
+The apollo-link's `Observable` follows the ECMAScript [proposal](https://github.com/tc39/proposal-observable#api) with some extensions provided by [zen-observable](https://github.com/zenparsing/zen-observable#api).
+
+### Creating a Simple Http Link
+
+All links extend the base class, `ApolloLink`, which contains the methods for composition as well as the signature of request.
```

Comment:
Good thought! It would make the `ApolloLink` into an interface. The reason to included `split` and `concat` on `ApolloLink` is to make long compositions in user code more readable/clean. Also users won't have to import ApolloLink if Links are used across different files.

I think this composition is more understandable than nesting `from`s and the `split`

```js
const link = loggingLink
.split(
  isLocal,
  checkLocalCache,
  ApolloLink.passthrough(),
)
.concat(dedup)
.concat(retry)
.concat(http)
```